wgpu-remote/src/lib.rs

```diff
@@ -0,0 +1,67 @@
+use ipc_channel::ipc::IpcSender;
+use parking_lot::Mutex;
+use serde::{Serialize, Deserialize};
+
+use wgpu_native as wgn;
+
+
+#[derive(Serialize, Deserialize)]
+pub enum InstanceMessage {
+    InstanceGetAdapter(wgn::InstanceId, wgn::AdapterDescriptor, wgn::AdapterId),
+    AdapterCreateDevice(wgn::AdapterId, wgn::DeviceDescriptor, wgn::DeviceId),
+}
+
+/// A message on the timeline of devices, queues, and resources.
+#[derive(Serialize, Deserialize)]
+pub enum GlobalMessage {
+    Instance(InstanceMessage),
+    //Device(DeviceMessage),
+    //Queue(QueueMessage),
+    //Texture(TextureMessage),
+    //Command(CommandMessage),
+}
+
+#[derive(Default)]
+pub struct IdentityHub {
+    adapters: wgn::IdentityManager,
+    devices: wgn::IdentityManager,
+}
+
+pub struct Client {
+    channel: IpcSender<GlobalMessage>,
+    identity: Mutex<IdentityHub>,
+}
+
+impl Client {
+    pub fn new(channel: IpcSender<GlobalMessage>) -> Self {
```

Comment:
We'll need to expose this in a C extern too, right?

Reply:
sure, at some point in the near future :)